### PR TITLE
EZP-31008: Moved repository-forms code to ezplatform-admin-ui package

### PR DESCRIPTION
**Pull request by webhdx**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-31008
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

*Reopening #1120 from `ezsystems` branch to fix Travis builds.*

# Description
I've moved in Content Type editing and Limitation Form/Value mappers as these were not used outside of AdminUI. This change introduces some BC breaks:

| Old location | New location |
|----------|-----|
|`EzSystems\RepositoryForms\FieldType\FieldDefinitionFormMapperInterface`|`EzSystems\EzPlatformAdminUi\FieldType\FieldDefinitionFormMapperInterface`|
|`EzSystems\RepositoryForms\Limitation\LimitationFormMapperInterface`|`EzSystems\EzPlatformAdminUi\Limitation\LimitationFormMapperInterface`|
|`EzSystems\RepositoryForms\Limitation\LimitationValueMapperInterface`|`EzSystems\EzPlatformAdminUi\Limitation\LimitationValueMapperInterface`|

All Action Dispatchers, Form Processors, Types and Data classes related to Content Types/Limitation are migrated too.

#### Compatibility note
Repository Forms will still provide BC layer through out 3.x release. You won't be affected if you are providing FieldTypes or Limitation Form/Value mappers. It might be required to align namespaces if you are using others parts from repository-forms package.

### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review
- [ ] Remove `[TMP]` commits before merge

### CODE REVIEW

File: src/bundle/Resources/config/ezplatform_default_settings.yaml [hunk 1]
```diff
@@ -48,6 +48,14 @@ parameters:
 
     # Views
     ezsettings.default.content_translate_view: {}
+    ezsettings.default.content_edit_view: {}
```
Comment (mikadamczyk) [inline src/bundle/Resources/config/ezplatform_default_settings.yaml]: Shouldn't we provide this configuration as a part of `ezsettings.admin_group`?

File: src/bundle/Resources/config/default_parameters.yaml [hunk 2]
```diff
@@ -1,5 +1,4 @@
 parameters:
-    ezsettings.admin_group.content_edit.templates.create_draft: '@@ezdesign/content/create.html.twig'
     ezsettings.admin_group.user_edit.templates.update: '@@ezdesign/user/edit.html.twig'
```
Comment (mikadamczyk) [inline src/bundle/Resources/config/default_parameters.yaml]: Maybe it is worth to move those lines to `src/bundle/Resources/config/ezplatform_default_settings.yaml`

File: src/bundle/Resources/config/services/twig.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        autoconfigure: true
+
+    EzSystems\EzPlatformAdminUiBundle\Templating\Twig\FieldEditRenderingExtension:
+        arguments: ["@ezpublish.templating.field_block_renderer"]
+        tags:
+            - { name: twig.extension }
+
+    EzSystems\EzPlatformAdminUiBundle\Templating\Twig\LimitationValueRenderingExtension:
+        arguments:
+            - '@EzSystems\EzPlatformAdminUi\Limitation\Templating\LimitationBlockRenderer'
```
Comment (mikadamczyk) [inline src/bundle/Resources/config/services/twig.yaml]: We should stick to one style to set arguments

File: src/bundle/Templating/Twig/FieldEditRenderingExtension.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+use eZ\Publish\Core\MVC\Symfony\Templating\FieldBlockRendererInterface;
+use EzSystems\EzPlatformAdminUi\Form\Data\FieldDefinitionData;
+use Twig\Environment;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class FieldEditRenderingExtension extends AbstractExtension
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\Templating\FieldBlockRendererInterface|\eZ\Publish\Core\MVC\Symfony\Templating\Twig\FieldBlockRenderer
```
Comment (mikadamczyk) [inline src/bundle/Templating/Twig/FieldEditRenderingExtension.php]: one line comment

File: src/bundle/Tests/DependencyInjection/Compiler/FieldTypeFormMapperDispatcherPassTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    protected function registerCompilerPass(ContainerBuilder $container): void
+    {
+        $container->addCompilerPass(new FieldTypeFormMapperDispatcherPass());
+    }
+
+    /**
+     * @dataProvider tagsProvider
+     */
+    public function testRegisterMappers(string $tag)
```
Comment (mikadamczyk) [inline src/bundle/Tests/DependencyInjection/Compiler/FieldTypeFormMapperDispatcherPassTest.php]:
```suggestion
    public function testRegisterMappers(string $tag): void
```

File: src/bundle/Tests/DependencyInjection/Compiler/LimitationValueMapperPassTest.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        $this->setDefinition(LimitationValueMapperPass::LIMITATION_VALUE_MAPPER_REGISTRY, new Definition());
+    }
+
+    protected function registerCompilerPass(ContainerBuilder $container): void
+    {
+        $container->addCompilerPass(new LimitationValueMapperPass());
+    }
+
+    public function testRegisterMappers()
```
Comment (mikadamczyk) [inline src/bundle/Tests/DependencyInjection/Compiler/LimitationValueMapperPassTest.php]:
```suggestion
    public function testRegisterMappers(): void
```

File: src/bundle/Tests/Templating/Twig/FieldEditRenderingExtensionTest.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+        return [new FieldEditRenderingExtension($fieldBlockRenderer)];
+    }
+
+    public function getFixturesDir()
+    {
+        return __DIR__ . '/_fixtures/field_edit_rendering_functions/';
+    }
+
+    public function getFieldDefinitionData($typeIdentifier, $id = null, $settings = [])
```
Comment (mikadamczyk) [inline src/bundle/Tests/Templating/Twig/FieldEditRenderingExtensionTest.php]:
```suggestion
    public function getFieldDefinitionData($typeIdentifier, $id = null, $settings = []): FieldDefinitionData
```

File: src/bundle/Tests/Templating/Twig/FieldEditRenderingExtensionTest.php [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        return new FieldDefinitionData([
+            'fieldDefinition' => new FieldDefinition([
+                'id' => $id,
+                'fieldSettings' => $settings,
+                'fieldTypeIdentifier' => $typeIdentifier,
+            ]),
+        ]);
+    }
+
+    private function getTemplatePath($tpl)
```
Comment (mikadamczyk) [inline src/bundle/Tests/Templating/Twig/FieldEditRenderingExtensionTest.php]:
```suggestion
    private function getTemplatePath($tpl): string
```

File: src/bundle/Tests/Templating/Twig/LimitationValueRenderingExtensionTest.php [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+                'priority' => 20,
+            ],
+        ]);
+
+        return [
+            new LimitationValueRenderingExtension($limitationBlockRenderer),
+        ];
+    }
+
+    private function createLimitationValueMapperRegistryMock()
```
Comment (mikadamczyk) [inline src/bundle/Tests/Templating/Twig/LimitationValueRenderingExtensionTest.php]: missing return typehint

File: src/bundle/Tests/Templating/Twig/LimitationValueRenderingExtensionTest.php [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+        $registryMock = $this->createMock(LimitationValueMapperRegistryInterface::class);
+        $registryMock
+            ->expects($this->atLeastOnce())
+            ->method('getMapper')
+            ->willReturn($mapperMock);
+
+        return $registryMock;
+    }
+
+    public function getLimitation($identifier, array $values)
```
Comment (mikadamczyk) [inline src/bundle/Tests/Templating/Twig/LimitationValueRenderingExtensionTest.php]: missing return typehint

File: src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+abstract class ConfigurableSudoRepositoryLoader
+{
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var array */
+    private $params;
+
+    /**
+     * @param \eZ\Publish\API\Repository\Repository $repository
```
Comment (mikadamczyk) [inline src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php]:
```suggestion

```

File: src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+        return $this->params[$name];
+    }
+
+    protected function getRepository(): Repository
+    {
+        return $this->repository;
+    }
+
+    /**
+     * @param \Closure $callback
```
Comment (mikadamczyk) [inline src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php]:
```suggestion

```

File: src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    public function setParam(string $name, $value): self
+    {
+        $this->params[$name] = $value;
+
+        return $this;
+    }
+
+    protected function getParam(string $name)
```
Comment (mikadamczyk) [inline src/lib/ConfigResolver/ConfigurableSudoRepositoryLoader.php]: `@return mixed` ?

File: src/lib/ConfigResolver/MaxUploadSize.php [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        return $value;
+    }
+
+    /**
+     * @param string $str
+     *
+     * @return int
+     */
```
Comment (mikadamczyk) [inline src/lib/ConfigResolver/MaxUploadSize.php]: This docblock in not needed

File: src/lib/Event/FieldDefinitionMappingEvent.php [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+    private $baseLanguage;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\Language|null */
+    private $targetLanguage;
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\Form\Data\FieldDefinitionData $fieldDefinitionData
+     * @param \eZ\Publish\API\Repository\Values\Content\Language|null $baseLanguage
+     * @param \eZ\Publish\API\Repository\Values\Content\Language|null $targetLanguage
+     */
```
Comment (mikadamczyk) [inline src/lib/Event/FieldDefinitionMappingEvent.php]: This docblock is not needed

File: src/lib/Event/FieldDefinitionMappingEvent.php [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+        ?Language $baseLanguage,
+        ?Language $targetLanguage
+    ) {
+        $this->baseLanguage = $baseLanguage;
+        $this->targetLanguage = $targetLanguage;
+        $this->fieldDefinitionData = $fieldDefinitionData;
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition
```
Comment (mikadamczyk) [inline src/lib/Event/FieldDefinitionMappingEvent.php]: This docblock is not needed

File: src/lib/Event/FieldDefinitionMappingEvent.php [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    public function getFieldDefinition(): FieldDefinition
+    {
+        return $this->fieldDefinitionData->fieldDefinition;
+    }
+
+    /**
+     * @return \EzSystems\EzPlatformAdminUi\Form\Data\FieldDefinitionData
```
Comment (mikadamczyk) [inline src/lib/Event/FieldDefinitionMappingEvent.php]: This docblock is not needed here and below

File: src/lib/EventListener/ViewTemplatesListener.php [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+{
+    /**
+     * Hash of [View type FQN] => template.
+     *
+     * @var array
+     */
+    protected $viewTemplates;
+
+    /**
+     * @var string
```
Comment (mikadamczyk) [inline src/lib/EventListener/ViewTemplatesListener.php]: one line comment

File: composer.json [hunk 19]
```diff
@@ -12,6 +12,24 @@
     "autoload-dev": {
         "psr-4": { "EzSystems\\EzPlatformAdminUi\\Tests\\": "src/lib/Tests" }
     },
+    "repositories": [
```
Comment (ViniTou) [inline composer.json]: I think this should not be here and also breaks travis

**Pull request merged**